htlcswitch/switch.go

```diff
@@ -566,37 +559,58 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 	// payment circuit by forwarding the settle msg to the channel from
 	// which htlc add packet was initially received.
 	case *lnwire.UpdateFufillHTLC, *lnwire.UpdateFailHTLC:
-		// Exit if we can't find and remove the active circuit to
-		// continue propagating the fail over.
-		circuit, err := s.circuits.remove(packet.payHash)
-		if err != nil {
-			err := errors.Errorf("unable to remove "+
-				"circuit for payment hash: %v", packet.payHash)
-			log.Error(err)
-			return err
-		}
+		if packet.incomingChanID == (lnwire.ShortChannelID{}) {
```

Comment:
Could use a comment here to indicate that the `packet.incomingChanID == (lnwire.ShortChannelID{})` fragment indicates the HTLC was initiated from a local subsystem. Alternatively, could be extracted out to a helper method on the packet as it reoccurs in a few places.

Reply:
It would seem now that the circuit is _only_ removed if the payment was initiated locally? If so this is a change in behavior and incorrect, as now the circuit map will never properly be garbage collected.